pkg/routeagent-driver/handlers/kubeproxy-iptables/endpoint_handler.go

```diff
@@ -0,0 +1,184 @@
+package kp_iptables
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/util"
+)
+
+func (kp *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	kp.localCableDriver = endpoint.Spec.Backend
+
+	// We are on nonGateway node
+	if endpoint.Spec.Hostname != kp.hostname {
+		// If the node already has a vxLAN interface that points to an oldEndpoint
+		// (i.e., during gateway migration), delete it.
+		if kp.vxlanDevice != nil && kp.vxlanDevice.activeEndpointHostname != endpoint.Spec.Hostname {
+			err := kp.vxlanDevice.deleteVxLanIface()
+			if err != nil {
+				return fmt.Errorf("failed to delete the the vxlan interface that points to old endpoint %s : %v",
+					kp.vxlanDevice.activeEndpointHostname, err)
+			}
+
+			kp.vxlanDevice = nil
+		}
+
+		kp.isGatewayNode = false
+		localClusterGwNodeIP := net.ParseIP(endpoint.Spec.PrivateIP)
+		remoteVtepIP, err := kp.getVxlanVtepIPAddress(localClusterGwNodeIP.String())
+		if err != nil {
+			return fmt.Errorf("failed to derive the remoteVtepIP %v", err)
+		}
+
+		klog.Infof("Creating the vxlan interface %s with gateway node IP %s", VxLANIface, localClusterGwNodeIP)
+		err = kp.createVxLANInterface(endpoint.Spec.Hostname, VxInterfaceWorker, localClusterGwNodeIP)
+		if err != nil {
+			klog.Fatalf("Unable to create VxLAN interface on non-GatewayNode (%s): %v", endpoint.Spec.Hostname, err)
+		}
+
+		kp.vxlanGwIP = &remoteVtepIP
+		err = kp.reconcileRoutes(remoteVtepIP)
+		if err != nil {
+			return fmt.Errorf("error while reconciling routes %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (kp *SyncHandler) LocalEndpointUpdated(endpoint *submV1.Endpoint) error {
+	return nil
+}
+
+func (kp *SyncHandler) LocalEndpointRemoved(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	kp.isGatewayNode = false
+
+	// If the vxLAN device exists and it points to the same endpoint, delete it.
+	if kp.vxlanDevice != nil && kp.vxlanDevice.activeEndpointHostname == endpoint.Spec.Hostname {
+		err := kp.vxlanDevice.deleteVxLanIface()
+		kp.vxlanDevice = nil
+		kp.vxlanGwIP = nil
+		if err != nil {
+			return fmt.Errorf("failed to delete the the vxlan interface on Endpoint removal: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (kp *SyncHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	if err := kp.overlappingSubnets(endpoint.Spec.Subnets); err != nil {
+		// Skip processing the endpoint when CIDRs overlap and return nil to avoid re-queuing.
+		klog.Errorf("IsoverlappingSubnets returned error: %v", err)
```

Comment:
It seems more context would be helpful?

```suggestion
		klog.Errorf("overlappingSubnets for new remote %#v returned error: %v", endpoint, err)
```